content/posts/2021-01-28-r-merging-datasets.es.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+
+```{r}
+adverse_events2 <- adverse_events
+colnames(adverse_events2)[1L] <- "pat_id"
+merge(
+  x = demographics,
+  y = adverse_events2,
+  by.x = "id",
+  by.y = "pat_id",
+  all = TRUE
+)
+```
+
+
+## Usando {dplyr}
+
+Al contrario que R {base}, que utiliza una sola función para realizar distintos tipos de fusión, {dplyr} tiene una función para cada tipo de fusión. Y afortunadamente sus nombre son tal cual se esperaría: `left_join()`,  `right_join()`, `inner_join()` y `full_join()` (del inglés izquierda, derecha, interna y completa, respectivamente). Personalmente soy un fan de este tipo de interfaz, y por ello tiendo a usar  {dplyr} más a menudo que  {base}.
+
+```{r, message=FALSE}
+library(dplyr)
+left_join(demographics, adverse_events, by = "id")
+inner_join(demographics, adverse_events, by = "id")
+full_join(demographics, adverse_events, by = "id")
+```
+
+En caso de que el nombre de las variables ID de las dos tablas no coincida, tendrás que pasar un vector nombrado como argumento de  `by`. Nombre y valor corresponden con la clave en la primera y segunda tabla, respectivamente.
+
+```{r}
+right_join(demographics, adverse_events2, by = c("id" = "pat_id"))
+```
+
+
+## Con SQL
+
+Cuando se trata de fusionar tablas, no hay más remedio que aludir a lenguaje de consultas structuradas (SQL). Hay varios paquetes de R disponibles en CRAN que sirven para enviar consultas SQL desde R a una base de datos. Sin embargo, el paquete  {tidyquery} hace algo diferente. Toma la consulta SQL que le das como entrada a  la función  `query()`, la traduce a código {dplyr} y ejecuta la secuencia  {dplyr} para producir el resultado final.
+
+```{r}
+library(tidyquery)
+query("select * from demographics right join adverse_events using(id)")
+query("select * from demographics inner join adverse_events using(id)")
+query("select * from demographics full join adverse_events using(id)")
+```
+
+Para consultas sencillas, tales como fusionar tablas, es posible que esto sea un exceso, dado que la interfaz de  {dplyr} es muy similar a SQL. Sin embargo, si eres un experto en  SQL y estás acostumbrado a hacer un uso más completo, {tidyquery}  puede ser un muy buen método para convertirse en experto en  {dplyr}, ya que te muestra el código traducido a {dplyr}.
+
+```{r}
+show_dplyr("
+  select dm.id, dm.age, ae.term
+  from demographics as dm
+  left join adverse_events as ae
+  using(id)
+  where term <> 'Headache'
+")
+```
+
+Por cierto, el paquete  {dbplyr} traduce tu código  {dplyr} a SQL. De este modo no necesitas en realidad aprender SQL para consultar una base de datos.
+
+En este artículo hemos visto los cuatro modos más habituales de fusionar tablas y cómo implementarlos en R utilizando {base}, {dpyr} y SQL vía {tidyquery}. Teniendo este conocimiento, deberías ser capaz de fusionar con confianza cualquier conjunto de datos que te encuentres en R. Si te atascas, no obstante, no dudes en preguntar en los comentarios aquí abajo.
```

Comment:
Please add a sentence saying "This article has been kindly translated by <your name or twitter handle with link to website/twitter/GitHub (whatever you like)>".

Reply:
I've just added the sentence at the end.